odoo/models.py

```diff
@@ -4112,13 +4100,60 @@ def _load_records(self, data_list, update=False):
         records = self._load_records_create([data['values'] for data in to_create])
         for data, record in zip(to_create, records):
             data['record'] = record
+            if data.get('xml_id'):
+                # add XML ids for parent records that have just been created
+                for parent_model, parent_field in self._inherits.items():
+                    if not data['values'].get(parent_field):
+                        imd_data_list.append({
+                            'xml_id': f"{data['xml_id']}_{parent_model.replace('.', '_')}",
+                            'record': record[parent_field],
+                            'noupdate': data.get('noupdate', False),
+                        })
+                imd_data_list.append(data)
 
         # create or update XMLIDs
-        imd_data_list = [data for data in data_list if data.get('xml_id')]
         imd._update_xmlids(imd_data_list, update)
 
         return original_self.concat(*(data['record'] for data in data_list))
 
+    def _log_item(self, items, item, *, created):
```

Comment:
What is the purpose of this method?

Reply:
Bit of a previous version I forgot to remove.